test/common_distributed.py

```diff
@@ -125,31 +125,52 @@ def setUpClass(cls):
                 fn = getattr(cls, attr)
                 setattr(cls, attr, cls.join_or_run(fn))
 
+    def __init__(self, *args, **kwargs):
+        super(MultiProcessTestCase, self).__init__(*args, **kwargs)
+
+        self.rank = self.MAIN_PROCESS_RANK
+
     def setUp(self):
         super(MultiProcessTestCase, self).setUp()
         self.skip_return_code_checks = []
-        self.rank = self.MAIN_PROCESS_RANK
-        self.file = tempfile.NamedTemporaryFile(delete=False)
-        self.processes = [self._spawn_process(rank) for rank in range(int(self.world_size))]
+        if self.rank == self.MAIN_PROCESS_RANK:
```

Comment:
Why do we need this change? Isn't the `setUp` method run only on the parent process?

Reply:
this was added to check whether setup() is related after tests failures, it did not help, but I think it is no harm to keep there, if child process want to set up same variables as parent in the future, they can call setup() function